keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/TensorShape.java

```diff
@@ -0,0 +1,31 @@
+package io.improbable.keanu.vertices.dbltensor;
+
+import java.util.Arrays;
+
+public class TensorShape {
```

Comment:
This feels a bit overkill if it only stores a 1d array. Can we not handle the logic for an array in `ProbabilisticVertexShaping`?

Reply:
It's just for the equals and hash code of an int array. Apparently Set<int[]> does exact comparison, which isn't what we need here. I've moved it into the `ProbabilisticVertexShaping` class because it's only used there.

Reply:
Nice